internal/pithosctl/pkg/cluster/config.go

```diff
@@ -19,67 +19,59 @@ package cluster
 import (
 	"encoding/base64"
 
+	"github.com/gravitational/pithos-app/internal/pithosctl/pkg/kubernetes"
+
 	"github.com/gravitational/trace"
 )
 
 // Config describes pithos application configuration
 type Config struct {
-	// KubeConfig defines path to kubernetes config file
-	KubeConfig string
+	// KubeClient defines kubernetes client
+	KubeClient *kubernetes.Client
 	// Namespace defines kubernetes namespace for pithos application components
 	Namespace string
 	// NodeSelector defines the filter for kubernetes nodes where cassandra should start
 	NodeSelector string
-	// Keys defines list of S3 keys which should be created during bootstrap
-	Keys []AccessKey
 	// CassandraPodSelector defines labels to select cassandra pods
 	CassandraPodSelector string
-	// Bootstrap represents bootstrapping configuration for pithos application
-	Bootstrap *Bootstrap
+	// PithosSecret defines secret name storing S3 keys
+	PithosSecret string
+	// ReplicationFactor defines replication factor for cassandra keyspace
+	// ReplicationFactor is ignored by pithos during upgrade and could be set to any value
```

Comment:
Is this a configuration that can be set externally or is it purely computed as required (as I've seen in the code)?

Also, not sure I understand the statement about caller's being able to set it to any value - does this simply mean that it is ignored during upgrades? Then I would definitely remove this bit as it is confusing.